03.bowling/frame.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+class Frame
+  attr_reader :shots
+
+  def initialize(shots, is_last_frame: false)
+    @shots = shots
+    @is_last_frame = is_last_frame
+  end
+
+  def score(index, all_frames)
+    base_score = @shots.sum(&:pins)
+    base_score = adjust_score_for_last_frame(base_score)
+    bonus = calculate_bonus(index, all_frames)
+    base_score + bonus
+  end
+
+  def strike?
+    @shots.first.strike?
+  end
+
+  def spare?
+    !strike? && @shots[0..1].sum(&:pins) == 10
```

Comment:
`sum(&:pins)`がたくさん出てくるので、

```suggestion
    !strike? && pins[0..1].sum == 10
```

みたいに書けるpinsメソッドがあるといいかも？